### DISCUSSION TITLE
ESP32: Online decoding of panic handler backtraces for stock firmware.

### DISCUSSION THREAD

**Discussion by RSC-Games** [micropython/micropython#15766] [Ideas]
Hello all!

I have noticed in a few recent issues I have commented on that the developers are running stock firmware and are unable to decode their backtrace. While I have access to a development machine during the weekends, other developers are unable to decode their backtrace, thus hindering the identification and fixing process.

In #9236, the MPY WLAN driver does not check whether it is in STA or AP mode, and therefore crashes when it tries to set an attribute it cannot set (like "dhcp-hostname"). The poster was unable to supply a decoded backtrace.

In #9176, soft rebooting a esp32 with a connected SDCard with non-standard pins causes a dereference on a null object. While the object is not known, it is guessed to be the SDCard object being cleaned up by the GC, and deinit causes the null dereference. Without a decoded backtrace, it is difficult to figure out more information.

It would be great if a micropython subdomain or webpage could be set up where any developer using stock firmware could decode their backtrace online. This would ease troubleshooting and allow us to figure out if the poster made a mistake or if there was an actual issue in micropython. This would only help with issues that don't involve corrupt stacks, but the only one that I saw involving corrupt stacks was my own, and that was my fault. (#8936)

If this is not possible please let me know.

**Comment by jonnor** [comment]
I assume track tracke here means an ESP32 stack trace? This would be useful for triaging. But as far as I understand, anyone could develop such a tool - and it could live on any domain and still be very useful? And if an existing tool exists, we could use that?
Contributions for such a thing would be welcomed. But I do not think that MicroPython core developers will be able to prioritize it anytime soon, as there are (literally) over 1000 other issues open that must be handled in MicroPython itself.

**Reply by RSC-Games** [reply]
Yeah. I am no longer actively working on the above project however and there are many existing tools that can be used on a standard development machine. I honestly had completely forgotten this issue was open.

I've moved away from ESP32 development onto PC-based projects (though for the latest one the ESP32 + MicroPython will make a return as a sort of system processor module).

**Comment by projectgus** [comment]
Actually we have a version of this now! Thanks to @DvdGiessen who put it together, when the esp32 port crashes then it now includes a link to this page: https://github.com/micropython/micropython/wiki/ESP32-debugging

This version still requires some local tools rather than being a webpage (which I agree would be great), however the instructions are pretty comprehensive now.
